provider/testdata/recorded/TestProviderUpgrade/cluster/1.0.4/state.json

```diff
@@ -0,0 +1,5309 @@
+                                    "server": "https://896889A17653C9C95E17AED52D6B09D7.gr7.us-west-2.eks.amazonaws.com"
+                                },
+                                "name": "kubernetes"
+                            }
+                        ],
+                        "contexts": [
+                            {
+                                "context": {
+                                    "cluster": "kubernetes",
+                                    "user": "aws"
+                                },
+                                "name": "aws"
+                            }
+                        ],
+                        "current-context": "aws",
+                        "kind": "Config",
+                        "users": [
+                            {
+                                "name": "aws",
+                                "user": {
+                                    "exec": {
+                                        "apiVersion": "client.authentication.k8s.io/v1beta1",
+                                        "args": [
+                                            "eks",
+                                            "get-token",
+                                            "--cluster-name",
+                                            "example-cluster-3-eksCluster-792a299"
+                                        ],
+                                        "command": "aws",
+                                        "env": [
+                                            {
+                                                "name": "KUBERNETES_EXEC_INFO",
+                                                "value": "{\"apiVersion\": \"client.authentication.k8s.io/v1beta1\"}"
+                                            },
+                                            {
+                                                "name": "AWS_PROFILE",
+                                                "value": "devsandbox"
+                                            }
+                                        ]
+                                    }
+                                }
+                            }
+                        ]
+                    }
+                },
+                "created": "2023-10-19T00:46:12.020491Z",
+                "modified": "2023-10-19T00:46:12.020491Z",
+                "sourcePosition": "project:///node_modules/@pulumi/runtime/stack.ts#36,23"
+            },
+            {
+                "urn": "urn:pulumi:p-it-antons-mac-cluster-0421759d::example-cluster::pulumi:providers:aws::default_5_31_0",
+                "custom": true,
+                "id": "4b6db602-50de-4739-ac98-93c6c721d82b",
+                "type": "pulumi:providers:aws",
+                "inputs": {
+                    "version": "5.31.0"
```

Comment:
Hmm. This is not right unfortunately. The specification stated 5.42.0 - which should have been the provider that got installed in PATH. Either that's the provider really being used but the version here is taken from the SDK, or the ambient plugin installation didn't work right.